src/compiler/checker.ts

```diff
@@ -19222,6 +19222,12 @@ namespace ts {
         }
 
         function isValidSpreadType(type: Type): boolean {
+            if (type.flags & TypeFlags.TypeVariable) {
+                const constraint = (<TypeVariable>type).constraint;

```

Comment:
The JSDoc comment of `TypeParameter#type` says: `/** Retrieve using getConstraintFromTypeParameter */`

Reply:
Right, had missed this. Thanks @ajafff.